File: app/src/main/java/com/wilinz/yuetingmusic/ui/playlist/PlaylistViewModel1.java
package com.wilinz.yuetingmusic.ui.playlist;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.wilinz.yuetingmusic.data.model.Song;
import com.wilinz.yuetingmusic.data.repository.SongRepository;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

public abstract class PlaylistViewModel1 extends AndroidViewModel {

    public PlaylistViewModel1(@NonNull Application application) {
        super(application);
    }

    public LiveData<List<Song>> getSongsLiveData() {
        return songsLiveData;
    }

    protected MutableLiveData<Boolean> refreshingLiveData=new MutableLiveData<>();
    protected MutableLiveData<List<Song>> songsLiveData = new MutableLiveData<>();

    public abstract Observable<List<Song>> getSongsList();

    public LiveData<Boolean> getRefreshingLiveData() {
        return refreshingLiveData;
    }
}

File: app/src/main/java/com/wilinz/yuetingmusic/ui/playlist/PlaylistViewModel1.kt
package com.wilinz.yuetingmusic.ui.playlist

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.wilinz.yuetingmusic.data.model.Song
import io.reactivex.rxjava3.core.Observable

abstract class PlaylistViewModel1(application: Application) : AndroidViewModel(application) {
    fun getSongsLiveData(): LiveData<List<Song>> {
        return songsLiveData
    }

    @JvmField
    var refreshingLiveData = MutableLiveData<Boolean>()
    @JvmField
    var songsLiveData = MutableLiveData<List<Song>>()
    abstract val songsList: Observable<List<Song>>
    fun getRefreshingLiveData(): LiveData<Boolean> {
        return refreshingLiveData
    }
}